src/firefly/js/util/WebUtil.js

```diff
@@ -413,32 +415,41 @@ export function downloadWithProgress(url, numTries=1000) {
     });
 }
 
-export function download(url) {
-    let nullFrame = document.getElementById('null_frame');
-    if (!nullFrame) {
-        nullFrame = document.createElement('iframe');
-        nullFrame.id = 'null_frame';
-        nullFrame.style.display = 'none';
-        nullFrame.style.width = '0px';
-        nullFrame.style.height = '0px';
-        document.body.appendChild(nullFrame);
+export function downloadBlob(blob, filename) {
+    if (blob) {
+        window.URL = window.URL || window.webkitURL;
+        const a = document.createElement('a');
+        a.href = window.URL.createObjectURL(blob);
+        a.download = filename;
+        a.click();
+        window.URL.revokeObjectURL(a.href);
     }
-    nullFrame.src = url;
-}
-
-export function downloadViaAnchor(url, filename) {
-    let downloadAnchor = document.getElementById('download_anchor');
-    if (!downloadAnchor) {
-        downloadAnchor = document.createElement('a');
-        downloadAnchor.id = 'download_anchor';
-        downloadAnchor.style.display = 'block';
-        downloadAnchor.style.width = '0px';
-        downloadAnchor.style.height = '0px';
-        document.body.appendChild(downloadAnchor);
+}
+
+function resolveFileName(resp) {
+    if (resp && resp.headers) {
+        const cd = resp.headers.get('Content-Disposition') || '';
+        const parts = cd.match(/.*filename=(.*)/);
+        if (parts && parts.length > 1) return parts[1];
     }
-    downloadAnchor.download = filename;
-    downloadAnchor.href = url;
-    downloadAnchor.click();
+}
+
+export function download(url, filename) {
+    const {protocol, host, path, hash, searchObject={}} = parseUrl(url);
+    const cmd = searchObject[ServerParams.COMMAND];
+    url = `${protocol}//${host}${path}?${ServerParams.COMMAND}=${cmd}` + (hash ? '#' + hash : '');      // add cmd into the url as a workaround for server-side code not supporting it
+
+    const params = Object.fromEntries(
+                        Object.entries(searchObject)
+                              .map(([k, v]) => [k, (isPlainObject(v) ? JSON.stringify(v) : v)]) );          // convert object back into JSON if needed.
+
+    fetchUrl(url, {method: 'post', params})
+        .then((resp) => {
+            filename = filename || resolveFileName(resp);
+            return resp.blob(); })
+        .then( (blob) => {
+            downloadBlob(blob, filename); })
```

Comment:
I am a little concerned about this step.  Maybe I am wrong but it appears that we don't stream a file out anymore but hold it in memory.  This might not work with a large FITS or table file. Have you tested that?

Reply:
It's not clear to me based on the documentation of Response.blob() that it will returns a fully populated content from the response or it'll return a blob backed by the response's input stream. 
I've tried a 1GB fits file and a 1 million rows table (800MB).  I don't see any increase in memory usage on the browser.  It looks like it's doing the later.
I did notice a BIG performance problem with TableSave when it's a big table.  The UI may need to change to account for a slow prep-stage as well.  I will create a new ticket for this.
@robyww is this acceptable?

Reply:
That's fine as long as you tested it.  I guess it could be backed my the stream.

Reply:
ticket created for UI changes when file is large.  https://jira.ipac.caltech.edu/browse/IRSA-2975